### ISSUE TITLE
null / undefined handling

### ISSUE THREAD

**Issue by TheYoxy** [Code-Forge-Net/remix-hook-form#105]
As is, when you create a form data with a value that has been defined with a value, even values that represents *nothing* (`null` and `undefined`), it gets serialized in string. 

Wouldn't it be better to ignore theses values or return the raw underlined type ?

One idea of test implementation in the case that the choice is to handle base type: 
```typescript
  describe('should handle undefined', () => {
    it('with stringify all', () => {
      const data = {
        undefined: undefined,
      };
      const formData = createFormData(data, true);
      expect(formData.get('undefined')).toEqual(undefined);
    });
    it('with stringify all', () => {
      const data = {
        undefined: undefined,
      };
      const formData = createFormData(data, false);
      expect(formData.get('undefined')).toEqual(undefined);
    });
  });
  describe('should handle null', () => {
    it('with stringify all', () => {
      const data = {
        null: null,
      };
      const formData = createFormData(data, true);
      expect(formData.get('null')).toEqual(null);
    });
    it('with stringify all', () => {
      const data = {
        null: null,
      };
      const formData = createFormData(data, false);
      expect(formData.get('null')).toEqual(null);
    });
  });
```

**Comment by theskillwithin** [comment]
+!

**Comment by theskillwithin** [comment]
@TheYoxy actually I don't think this can be done in `createFormData` because this is using `FormData`

> https://developer.mozilla.org/en-US/docs/Web/API/FormData/append#value

What is really desired here is something that conform does with parseWithZod

https://conform.guide/integration/remix

**Comment by hannesj** [comment]
Null works already when used together with (`generateFormData`)[https://github.com/forge42dev/remix-hook-form/blob/main/src/utilities/index.ts#L31], as `JSON.parse("null")` returns `null`. However, `undefined` is not valid JSON, so the function throws, and returns `"undefined"` instead. I think `createFormData` should just filter out undefined values, as `formData.get` would then return `undefined`.

**Comment by oezbeyyf** [comment]
That's a good point regarding "null". So basically, https://github.com/theskillwithin/remix-hook-form/pull/1 should do what we want (filter "undefined" in createFormData before putting the data into FormData and sending it). 
But perhaps the same check for "null" should be disregarded, as "null" should properly work with JSON.parse("null")?

**Comment by AlemTuzlak** [comment]
@here fixed with 5.1.0, from this version onwards undefined values are ignored and not sent over the wire as they are not really useful in any way, null values are still serialized and the reason behind htis is that sending null to your database might mean a whole different thing than just not sending anything so you might want the null to be there to remove some set values. I would recommend handling the null values in your own code if you want to ignore them
